File: app/src/main/java/com/github/goldy1992/mp3player/service/library/content/searcher/ContentSearcher.java
package com.github.goldy1992.mp3player.service.library.content.searcher;

import android.support.v4.media.MediaBrowserCompat.MediaItem;

import androidx.annotation.NonNull;

import com.github.goldy1992.mp3player.commons.MediaItemType;

import java.util.List;

public interface ContentSearcher {

    List<MediaItem> search(@NonNull String query);
    MediaItemType getSearchCategory();

}

File: android/service/src/main/java/com/github/goldy1992/mp3player/service/library/content/searcher/ContentSearcher.kt
package com.github.goldy1992.mp3player.service.library.content.searcher

import android.support.v4.media.MediaBrowserCompat
import com.github.goldy1992.mp3player.commons.MediaItemType

interface ContentSearcher {
    fun search(query: String): List<MediaBrowserCompat.MediaItem>?
    val searchCategory: MediaItemType?
}